query-node/joystream-query-node/README.md

```diff
@@ -24,6 +24,81 @@ $ cd generated/indexer
 $ yarn start
 ```
 
+## Add a new mapping for Joystream MemberRegistered event
+
+1. Every mapping function get a parameter of `QueryEvent` type
+
+```ts
+import { QueryEvent, db } from "../generated/indexer/index";
+```
+
+2. `db` object for database operations: save/get/remove
+
+```ts
+import { db } from "../generated/indexer/index";
+```
+
+3. Define the event handler function with the following signature
+
+```ts
+export async function handleMemberRegistered(event: QueryEvent) {}
+```
+
+4. Inside the handler function create a new instance of the entity and fill properties with event data.
+
+```ts
+const member = new MemberRegistereds();
+// Get event data
+const { AccountId, MemberId } = event.event_params;
+
+member.accountId = AccountId;
+member.memberId = +MemberId;
+```
+
+5. Call `db.save()` method to save data on database
+
+```ts
+// Save to database.
+db.save(member, event);
+```
+
+6. Query database
+
+```ts
+// Query from database
+const m = await db.get("MemberRegistereds", { memberId: MemberId });
+console.log(m);
+```
+
+**Notes**
+`db.save()` method take two parameters entity instance and event. `db.save()` method use `event` to keep track of last processed event.
+
+`db.get()` method. The first parameter is the name of the entity class to query and the second parameter is the find options (to match record).
+
+**Complete code**
+
+```ts
+import { MemberRegistereds } from "../generated/indexer/entities/MemberRegistereds";
```

Comment:
This example needs to be tweaked a bit.

First off, we really want the query to be for memberships, not registration events. We could also want to index registration events, but I would say that is secondary. So the name of the type should be `Membership`. We process the `MembershipRegistered` (not `MemberRegistereds` as you have written) runtime event to add new entities to the `Membership` table. For completeness, you may as well include one other related event, for example, one that has to do with editing the profile of an event, or something. That way we have two separate events that have to do with providing one query into the memberships on the platform.